File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        u = len(nums) - 1
        l = 0
        while(l<=u):
            m = l + ((u - l) // 2)
            if target < nums[m]:
                u = m - 1
            elif target > nums[m]:
                l = m +1
            else:
                return m
        return -1   

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int s;
        int n = numbers.size();
        for (int i = 0; i <n; i++) {
            int curr = numbers[i];
            int j = i+1;
            while(j<n) {
                int fut = numbers[j];
                if ((curr+fut) > target)
                    break;
                else if ((curr+fut)==target)
                    return {i+1,j+1};
                j+=1;
            }
        }
        return numbers;
    }
};
